state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -306,6 +306,22 @@ impl<T: Config> Swap<T> {
 	}
 }
 
+#[derive(Debug, Decode, TypeInfo, Clone, PartialEq, Eq, Encode)]
+pub enum SwapFailureReason {
+	// Batch swap failed due to price impact limit,
```

Comment:
nit: why not `///` doc comments ?

further nit: punctuation is a little weird/inconsistent... trailing comma makes the comments look incomplete.